server/src/main/java/org/elasticsearch/action/admin/indices/sampling/SamplingConfiguration.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.action.admin.indices.sampling;
+
+import org.elasticsearch.cluster.Diff;
+import org.elasticsearch.cluster.SimpleDiffable;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+
+import static org.elasticsearch.xcontent.ConstructingObjectParser.constructorArg;
+import static org.elasticsearch.xcontent.ConstructingObjectParser.optionalConstructorArg;
+
+/**
+ * Configuration for sampling raw documents in an index.
+ *
+ * rate (required): The fraction of documents to sample (between 0 and 1).
+ * maxSamples (optional): The maximum number of documents to sample.
+ * maxSize (optional): The maximum total size of sampled documents.
+ * timeToLive (optional): The duration for which the sampled documents should be retained.
+ * condition (optional): An optional condition that sampled documents must satisfy.
+ */
+public record SamplingConfiguration(double rate, Integer maxSamples, ByteSizeValue maxSize, TimeValue timeToLive, String condition)
+    implements
+        ToXContentObject,
+        SimpleDiffable<SamplingConfiguration> {
+
+    public static final String TYPE = "sampling_configuration";
+    private static final String RATE_FIELD_NAME = "rate";
+    private static final String MAX_SAMPLES_FIELD_NAME = "max_samples";
+    private static final String MAX_SIZE_FIELD_NAME = "max_size";
+    private static final String TIME_TO_LIVE_FIELD_NAME = "time_to_live";
+    private static final String CONDITION_FIELD_NAME = "condition";
```

Comment:
In the design doc, this field is `if` rather than `condition`, to be consistent with the analogous usage in ingest pipelines. If you want to make a case to change it, it's probably worth discussing in there.